src/journal/journalctl.c

```diff
@@ -1203,6 +1203,15 @@ static int get_boots(
                                 break;
                         }
                 } else {
+                        i = 0;
+                        LIST_FOREACH(boot_list, id, head) {
+                                if (sd_id128_equal(id->id, current->id)) {
+                                        /* boot id already stored, something wrong with the journal files */
+                                        /* exiting as otherwise this problem would cause forever loop */
+                                        goto finish;
+                                }
+                                i++;
```

Comment:
The code seems sane to me, except that the `id` variable could be local to that new `if` branch. As I am not familiar with the in-depth details of the journal implementation, I'd like to get a final Ack from @poettering.